server/services/apiService.js

```diff
@@ -0,0 +1,213 @@
+                response.status(STATUS_CODE_FAIL).
+                send({ error: errorMessage });
+            }
+        })
+    });
+}
+
+function changeArchiveType (title, response) {
+    databaseService.getRowByTitle(SUBREDDITS_TABLE_TITLE, title).then((subreddit) => {
+        const isArchived = subreddit ? subreddit.isArchived : 0;
+        const newValue = isArchived === 0 ? 1 : 0;
+
+        databaseService.updateTable(SUBREDDITS_TABLE_TITLE, 'isArchived', subreddit.id, newValue).then(resolve => {
+            const subreddits = databaseService.getDataFromDatabase(SUBREDDITS_TABLE_TITLE);
+            subreddits.then(subreddits => {
+                response.status(STATUS_CODE_OK).send(subreddits);
+            });
+
+            if (newValue) startRedditArchivation(subreddit.display_name, subreddit.id);
+            if (!newValue) removePostsAndComments(subreddit.display_name);
+        }).catch(() => {
+            response.send(STATUS_CODE_FAIL)
+        })
+    })
+}
+
+function removePostsAndComments (subredditName) {
+    databaseService.removeRowsFromTable(POSTS_TABLE_TITLE, 'subreddit', subredditName);
+    databaseService.removeRowsFromTable(COMMENTS_TABLE_TITLE, 'subreddit', subredditName);
+}
+
+function getSimilarSubredditNameFromAPI (subreddit) {
+    return redditFetcher.checkIsSubredditExists(subreddit).then(result => {
+        if (result.length && result[0].subreddit && result[0].subreddit.display_name) {
+            return result[0].subreddit.display_name
+        }
+        return false
+    }).catch(() => false);
+}
+
+function getSimilarSubredditFromDatabase (subredditTitle) {
+    if (subredditTitle) {
+        return databaseService.getDataFromDatabase(SUBREDDITS_TABLE_TITLE).then((result => {
+            const subreddit = result.filter(el => el.display_name.toLowerCase() === subredditTitle.toLowerCase());
+            return subreddit.length ? subreddit[0] : null
+        }))
+    }
+    return null
+}
+
+function addNewSubredditDatabase (subreddit, subredditId) {
+    const subredditObj = {
+        title: subreddit,
+        display_name: subreddit,
+        isArchived: 1
+    };
+    databaseService.insertDataInTable(SUBREDDITS_TABLE_TITLE, subredditObj);
+    startRedditArchivation(subreddit.display_name, subredditId);
+}
+
+function startRedditArchivation (subredditTitle, subredditId) {
+    redditFetcher.getSubredditPosts(subredditTitle).then(result => {
+        if (result && result.length) {
+            result.forEach((post, index) => {
+                databaseService.insertDataInTable(POSTS_TABLE_TITLE, post, subredditId, index);
+                redditFetcher.fetchPostSubmission(post.subreddit.display_name, post.id).then(el => {
+                    const comments = getAllComments(el[1].data.children);
+                    comments.forEach((comment) => {
+                        if (comment) {
+                            comment.data.postId = post.id;
```

Comment:
Function with a lot of nesting are quite hard to maintain, usually you should have no more than 2-3 levels of nesting in 1 function. You can use early return to reduce nesting, or extract some code into separate functions

Reply:
fixed https://github.com/YuriyBC/Reddit-archive/commit/c18789b84ca78579c468d22cc167eb365b736a29